src/main/java/apoc/util/FileUtils.java

```diff
@@ -31,6 +32,7 @@
     public static final String HDFS_PROTOCOL = "hdfs";
     public static final boolean HDFS_ENABLED = Util.classExists("org.apache.hadoop.fs.FileSystem");
     public static final Pattern HDFS_PATTERN = Pattern.compile("^(hdfs:\\/\\/)(?:[^@\\/\\n]+@)?([^\\/\\n]+)");
+    public static final Pattern S3_PATTERN = Pattern.compile("^(s3:\\/\\/)(?:[^@\\/\\n]+@)?([^\\/\\n]+)");
```

Comment:
do you think makes sense have some kind of factory pattern here and use it instead the `if/else/if`?